src/include/btree.h

```diff
@@ -130,6 +126,30 @@ struct __wt_btree {
 	uint64_t rec_max_txn;		/* Maximum txn seen (clean trees) */
 	uint64_t write_gen;		/* Write generation */
 
+	enum {
+		WT_CKPT_OFF, WT_CKPT_PREPARE, WT_CKPT_RUNNING
+	} checkpointing;		/* Checkpoint in progress */
+
+	/*
+	 * We flush pages from the tree (in order to make checkpoint faster),
+	 * without a high-level lock.  To avoid multiple threads flushing at
+	 * the same time, lock the tree.
+	 */
+	WT_SPINLOCK	flush_lock;	/* Lock to flush the tree's pages */
+
+	bool modified;			/* If the tree ever modified */
+	bool bulk_load_ok;		/* Bulk-load is a possibility */
+
+	/*
+	 * The tree's cache and eviction information persist past after the
+	 * handle is closed (clean cache pages may remain after the tree is
+	 * closed). Be careful clearing the WT_BTREE structure.
+	 */
+#define	WT_BTREE_CLEAR_SIZE(b)						\
+	(WT_PTRDIFF(&(b)->root, b))
```

Comment:
I think it's clearer to use:

`#define	WT_BTREE_CLEAR_SIZE offsetof(WT_BTREE, root)`

I note that `WT_SESSION_CLEAR_SIZE` uses the same kind of calculation.

The bigger issue is why do we need to keep the fields below on a closed tree (particularly since they will be cleared on next open)?  Ideally we could free `dhandle->handle` in that case -- can't we just assert that eviction isn't active in a closed tree?